test/test_optim.py

```diff
@@ -477,6 +477,88 @@ def test_multi_tensor_optimizers(self):
             for p1, p2 in zip(res[0], res[1]):
                 self.assertEqual(p1, p2, atol=5e-5, rtol=0)
 
+    def test_optimizers_foreach_flag(self):
+        if not torch.cuda.is_available():
+            return
+
+        optimizer_pairs_with_flags = [
+            # ((optim.Adam), dict(weight_decay=1., amsgrad=True)),
+            # ((optim.Adam), dict(weight_decay=1., amsgrad=False)),
+            # ((optim.Adam), dict(weight_decay=0., amsgrad=True)),
+            # ((optim.Adam), dict(weight_decay=0., amsgrad=False)),
+            # ((optim.AdamW), dict(weight_decay=1., amsgrad=True)),
+            # ((optim.AdamW), dict(weight_decay=1., amsgrad=False)),
+            # ((optim.AdamW), dict(weight_decay=0., amsgrad=True)),
+            # ((optim.AdamW), dict(weight_decay=0., amsgrad=False)),
+            # ((optim.NAdam), dict(weight_decay=0., momentum_decay=6e-3)),
+            # ((optim.NAdam), dict(weight_decay=1., momentum_decay=6e-3)),
+            # ((optim.NAdam), dict(weight_decay=0., momentum_decay=4e-3)),
+            # ((optim.NAdam), dict(weight_decay=0.01, momentum_decay=4e-3)),
+            # ((optim.SGD), dict(lr=0.2, momentum=1, dampening=0, weight_decay=1, nesterov=True)),
+            # ((optim.SGD), dict(lr=0.2, momentum=1, dampening=0.5, weight_decay=1, nesterov=False)),
+            # ((optim.RAdam), dict(weight_decay=0)),
+            # ((optim.RAdam), dict(weight_decay=1)),
+            # ((optim.RMSprop), dict(weight_decay=1, momentum=1, centered=True)),
+            # ((optim.RMSprop), dict(weight_decay=1, momentum=0, centered=True)),
+            # ((optim.RMSprop), dict(weight_decay=1, momentum=1, centered=False)),
+            # ((optim.RMSprop), dict(weight_decay=0, momentum=1, centered=False)),
+            # ((optim.Rprop), dict(lr=1e-2, etas=(0.5, 1.2), step_sizes=(1e-6, 50))),
+            # ((optim.ASGD), dict(weight_decay=0)),
+            # ((optim.ASGD), dict(weight_decay=1)),
+            # ((optim.Adamax), dict(weight_decay=0)),
+            # ((optim.Adamax), dict(weight_decay=1)),
+            ((optim.Adadelta), dict(weight_decay=0)),
+            ((optim.Adadelta), dict(weight_decay=1)),
+            # ((optim.Adagrad), dict(weight_decay=0)),
+            # ((optim.Adagrad), dict(weight_decay=1)),
+        ]
+
+        kIterations = 11
+        device = 'cuda'
+
+        for opt, params in optimizer_pairs_with_flags:
+            res = []
+            for foreach in (True, False):
+                weight = torch.tensor([[-0.2109, -0.4976], [-0.1413, -0.3420], [-0.2524, 0.6976]],
```

Comment:
Why hardcode values here instead of torch.rand?
If you want the same values in both loops, you can create these outside.